src/CppUTestExt/MockSupport_c.cpp

```diff
@@ -32,6 +32,8 @@
 #include "CppUTestExt/MockSupport.h"
 #include "CppUTestExt/MockSupport_c.h"
 
+typedef void Cpp_Func();  /* Cl2000 requires casts to C++ function */
+
```

Comment:
Yes, but that's besides the point.

Tthe point is that the cast must be __to a c++ function__. I dont'know any syntactically correct way to do that inline.

The typedef works, because it is in the context of C++ code. A typdef could also be put with in extern "C++" {}.